PurchasesCoreSwift/Attribution/AttributionPoster.swift

```diff
@@ -0,0 +1,167 @@
+    }
+
+    @objc(postAttributionData:fromNetwork:forNetworkUserId:)
+    public func post(attributionData data: [String: Any],
+                     fromNetwork network: AttributionNetwork,
+                     forNetworkUserId networkUserId: String?) {
+        Logger.debug(Strings.attribution.instance_configured_posting_attribution)
+        if data["rc_appsflyer_id"] != nil {
+            Logger.warn(Strings.attribution.appsflyer_id_deprecated)
+        }
+
+        if network == .appsFlyer && networkUserId == nil {
+            Logger.warn(Strings.attribution.networkuserid_required_for_appsflyer)
+        }
+
+        guard let appUserID = self.identityManager.maybeCurrentAppUserID else {
+            Logger.error(Strings.attribution.missing_app_user_id)
+            return
+        }
+
+        let identifierForAdvertisers = attributionFetcher.identifierForAdvertisers
+        if identifierForAdvertisers == nil {
+            Logger.warn(Strings.attribution.missing_advertiser_identifiers)
+        }
+
+        let networkKey = String(network.rawValue)
+        let dictOfLatestNetworkIdsAndAdvertisingIdsSentToNetworks = deviceCache.latestNetworkAndAdvertisingIdsSent(appUserID: appUserID)
+        let latestSentToNetwork = dictOfLatestNetworkIdsAndAdvertisingIdsSentToNetworks[networkKey]
+
+        // TODO: `(null)` is true to the ObjC code here, maybe we should reject this and not post?
+        let newValueForNetwork = "\(identifierForAdvertisers ?? "(null)")_\(networkUserId ?? "(null)")"
+        guard latestSentToNetwork != newValueForNetwork else {
+            Logger.debug(Strings.attribution.skip_same_attributes)
+            return
+        }
+
+        var newDictToCache = dictOfLatestNetworkIdsAndAdvertisingIdsSentToNetworks
+        newDictToCache[networkKey] = newValueForNetwork
+        var newData = data
+
+        newData["rc_idfa"] = identifierForAdvertisers
+        newData["rc_idfv"] = attributionFetcher.identifierForVendor
+        newData["rc_attribution_network_id"] = networkUserId
+
+        if newData.count > 0 {
+            if network == .appleSearchAds {
+                backend.post(attributionData: newData, network: network, appUserID: appUserID) { maybeError in
+                    guard maybeError == nil else {
+                        return
+                    }
+
+                    self.deviceCache.set(latestNetworkAndAdvertisingIdsSent: newDictToCache, appUserID: appUserID)
+                }
+            } else {
+
+                self.subscriberAttributesManager
+                    .convertAttributionDataAndSetAsSubscriberAttributes(attributionData: newData,
+                                                                        network: network,
+                                                                        appUserID: appUserID)
+
+                self.deviceCache.set(latestNetworkAndAdvertisingIdsSent: newDictToCache, appUserID: appUserID)
+            }
```

Comment:
FFTI: maybe we could extract this into a postSearchAds and a postSubscriberAttributes methods so the if-else are one-liners?

Reply:
💯